lib/cocoapods/installer/analyzer.rb

```diff
@@ -860,6 +860,8 @@ def determine_platform(specs, target_definitions, build_type)
           minimum = Version.new('8.0')
           deployment_target = [deployment_target, minimum].max
         end
+        # Set deployment_target to max of the Podfile and podspec specifications.
+        deployment_target = [deployment_target, target_definitions.first.platform.deployment_target].max
```

Comment:
hmmm why the `first` target definition? I guess it doesnt matter because if a pod is embedded between two target definitions with different deployment targets then we split the pod target into each deployment target?

Reply:
Yes this function is already making that assumption.  See line 854.